addons/mass_mailing/models/mailing.py

```diff
@@ -1306,36 +1315,117 @@ def _unsubscribe_token(self, res_id, email):
         token = (self.env.cr.dbname, self.id, int(res_id), tools.ustr(email))
         return hmac.new(secret.encode('utf-8'), repr(token).encode('utf-8'), hashlib.sha512).hexdigest()
 
-    def _convert_inline_images_to_urls(self, body_html):
-        """
-        Find inline base64 encoded images, make an attachement out of
-        them and replace the inline image with an url to the attachement.
-        """
 
-        def _image_to_url(b64image: bytes):
-            """Store an image in an attachement and returns an url"""
-            attachment = self.env['ir.attachment'].create({
-                'datas': b64image,
-                'name': "cropped_image_mailing_{}".format(self.id),
-                'type': 'binary',})
+    def _image_to_url(self, b64image: bytes):
+        """Store an image in an attachement and returns an url"""
+        attachment = self.env['ir.attachment'].create({
+            'datas': b64image,
+            'name': "cropped_image_mailing_{}".format(self.id),
+            'type': 'binary',})
+
+        attachment.generate_access_token()
 
-            attachment.generate_access_token()
+        return '/web/image/%s?access_token=%s' % (
+            attachment.id, attachment.access_token)
 
-            return '/web/image/%s?access_token=%s' % (
-                attachment.id, attachment.access_token)
+    def _get_image_by_url(self, url, session):
+        maxsize = int(tools.config.get("import_image_maxbytes", DEFAULT_IMAGE_MAXBYTES))
```

Comment:
Do we really want to add a configuration for that ? What is the use case ? And why not a config parameter instead ?

Reply:
See below.